src/webview/src/components/textFieldEditDialog.js

```diff
@@ -0,0 +1,72 @@
+// copyright (c) 2020, Matthias Behr
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import { Button, DialogContent, DialogTitle, IconButton, TextField } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+
+/**
+ * open a modal dialog to edit text field information
+ * @param {*} props (open, onChange, onClose)
+ */
+export default function TextFieldEditDialog(props) {
+
+    console.log(`TextFieldEditDialog(open=${props.open})`);
+
+    const [dataTextValue, setDataTextValue] = React.useState();
+    const [dataRenderFormat, setRenderFormat] = React.useState();
+
+    useEffect(() => {
+        setDataTextValue(props.data?.textValue);
+        setRenderFormat(props.data?.renderFormat);
+    }, [props.data], props.placeholder, props.label);
```

Comment:
I think props.placeholder and label belong inside the [].
,[props.data, props.placeholder, props.label]);